test/dns/resolver.cpp

```diff
@@ -0,0 +1,39 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+
+#define CATCH_CONFIG_MAIN
+#include "../src/libmeasurement_kit/ext/catch.hpp"
+
+#include "../src/libmeasurement_kit/dns/resolver.hpp"
+
+using namespace mk;
+using namespace mk::dns;
+
+TEST_CASE("the system resolver is able to resolve an ipv4 address") {
+    loop_with_initial_event([]() {
+            system_resolver("IN", "A", "www.neubot.org", [](Error e, Var<Message> message) {
+                    REQUIRE(!e);
+                    REQUIRE(message->answers.size() == 1);
+                    REQUIRE(message->answers[0].ipv4 == "130.192.16.172");
+                    break_loop();
+                    }, {}, Reactor::global(), Logger::global());
+            });
+}
+
+TEST_CASE("the system resolver is able to resolve and ipv6 address") {
+    loop_with_initial_event([]() {
+            system_resolver("IN", "AAAA", "google.com", [](Error e, Var<Message> message) {
+                    REQUIRE(!e);
+                    REQUIRE(message->answers.size() > 0);
+                    auto found = false;
+                    for (auto answer : message->answers) {
+                        if (answer.ipv6 == "2a00:1450:4001:815::200e") {
+                                found = true;
+                        }
+                    }
+                    REQUIRE(found);
+                    break_loop();
+                    }, {}, Reactor::global(), Logger::global());
+            });
+}
```

Comment:
If we can mock the various call, I think we can also be satisfied with full coverage with IPv6 failing on travis (and we can perhaps add logic to deal with failures in the travis case?)